python/tank/util/shotgun.py

```diff
@@ -312,25 +324,121 @@ def create_sg_connection(user="default"):
     :param user: Optional shotgun config user to use when connecting to shotgun, as defined in shotgun.yml
     :returns: SG API instance
     """
-    api_handle, _ = __create_sg_connection(__get_sg_config(), evaluate_script_user=False, user=user)
+    # The "user" parameter was introduced by Psyop for Psyop. We will be supporting it in legacy
+    # code paths, but not in new ones.
+    if user != "default":
+        config_data = __get_sg_config_data(__get_sg_config(), user)
+        api_handle = __create_sg_connection(config_data)
+    else:
+        api_handle = __create_sg_connection()
     return api_handle
 
+g_app_store_connection = None
 def create_sg_app_store_connection():
     """
-    Creates a shotgun connection to the tank app store.
-
-    returns a tuple: (api_handle, script_user_entity)
+    Creates a shotgun connection that can be used to access the Toolkit app store.
 
-    The second part of the tuple represents the
-    user that was used to connect to the app store,
-    as a standard sg entity dictionary.
+    :returns: (sg, dict) where the first item is the shotgun api instance and the second 
+              is an sg entity dictionary (keys type/id) corresponding to to the user used
+              to connect to the app store.
     """
     global g_app_store_connection
+
+    # maintain a cache for performance
+    if g_app_store_connection is not None:
+        return g_app_store_connection
+
+    config_data = __get_app_store_connection_information()
```

Comment:
Would be nice with a comment here saying "connect to associated shotgun site and retrieve the credentials to use to connect to the app store site" or something along those lines.